databricks/koalas/indexes.py

```diff
@@ -565,6 +567,8 @@ def name(self) -> Union[Any, Tuple]:
 
     @name.setter
     def name(self, name: Union[Any, Tuple]) -> None:
+        if not is_hashable(name):
+            raise TypeError("Index.name must be a hashable type")
```

Comment:
We should check this in `names` setter?
E.g.,:

```py
>>> pidx = pd.Index([1, 2, 3])
>>> pidx.names = [["0"]]
Traceback (most recent call last):
...
TypeError: Int64Index.name must be a hashable type
```

Reply:
Thanks for the comments!! 👍